addons/website/static/src/snippets/s_image_gallery/options.js

```diff
@@ -32,12 +32,44 @@ options.registry.gallery = options.Class.extend({
             return layoutPromise.then(this._super.apply(this, arguments));
         }
 
-        // Make sure image previews are updated if images are changed
-        this.$target.on('image_changed.gallery', 'img', function (ev) {
-            var $img = $(ev.currentTarget);
-            var index = self.$target.find('.carousel-item.active').index();
-            self.$('.carousel:first li[data-bs-target]:eq(' + index + ')')
-                .css('background-image', 'url(' + $img.attr('src') + ')');
+        // Manage the media replacement. Make sure the data-index attribute and
+        // the media previews are updated if medias are changed.
+        // TODO init_media_dialog is not triggered anymore
```

Comment:
I added this comment here as a reminder, not sure if there is an appropriate place in 16.0 to trigger this event. Maybe there is a better way in 16.0 who knows :shrug: I let you check :+1: